### ISSUE TITLE
[openstack method] Playbook fails in openshift-cluster/list.yml

### ISSUE THREAD

**Issue by dougbtv** [openshift/openshift-ansible#2856]
Playbook fails with `the field 'args' has an invalid value, which appears to include a variable that is undefined` when using the openstack cluster create script. I believe this is potentially related to [this commit](https://github.com/openshift/openshift-ansible/commit/82449c6b8a27ef56db70af2d8c0be89dc24465d2) authored by @lhuard1A which adds the [failing lines here](https://github.com/openshift/openshift-ansible/commit/82449c6b8a27ef56db70af2d8c0be89dc24465d2#diff-1103d42e01f80d3ebb80172dae5f89deR20) to the `playbooks/openstack/openshift-cluster/list.yml`

##### Version
current head (openshift-ansible-3.4.17-1-158-g90fb07f)
ansible 2.2.0.0

##### Steps To Reproduce
1. use openstack method and deploy like:

```
[stack@undercloud openshift-ansible]$ bin/cluster create   -o image_name=centos-custom   -o external_net=ext-net   -o floating_ip_pool=ext-net   -o net_cidr=40.0.0.0/24   openstack test_cluster
```

##### Current Result

The `add_host` task fails when looking for a variable in the `playbooks/openstack/openshift-cluster/list.yml` playbook.

```
TASK [add_host] ****************************************************************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'dict object' has no attribute 'openstack'\n\nThe error appears to have been in '/home/stack/openshift-ansible/playbooks/openstack/openshift-cluster/list.yml': line 14, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n    when: cluster_id == ''\n  - add_host:\n    ^ here\n"}

NO MORE HOSTS LEFT *************************************************************
  to retry, use: --limit @/home/stack/openshift-ansible/playbooks/openstack/openshift-cluster/launch.retry

PLAY RECAP *********************************************************************
localhost                  : ok=35   changed=21   unreachable=0    failed=1   
test_cluster-master-0      : ok=444  changed=117  unreachable=0    failed=0   
test_cluster-node-compute-0 : ok=176  changed=57   unreachable=0    failed=0   
test_cluster-node-compute-1 : ok=170  changed=56   unreachable=0    failed=0   
test_cluster-node-infra-0  : ok=170  changed=56   unreachable=0    failed=0   

ACTION [create] failed: Command 'ansible-playbook  -i inventory/openstack/hosts -e 'num_masters=1 cli_external_net=ext-net cli_net_cidr=40.0.0.0/24 cli_floating_ip_pool=ext-net num_nodes=2 cluster_id=test_cluster cluster_env=dev cli_image_name=centos-custom num_etcd=0 num_infra=1 deployment_type=origin' playbooks/openstack/openshift-cluster/launch.yml' returned non-zero exit status 2
[stack@undercloud openshift-ansible]$ 

```

The play in question is:

```
---
- name: Generate oo_list_hosts group
  hosts: localhost
  become: no
  connection: local
  gather_facts: no
  vars_files:
  - vars.yml
  tasks:
  - set_fact: scratch_group=meta-clusterid_{{ cluster_id }}
    when: cluster_id != ''
  - set_fact: scratch_group=all
    when: cluster_id == ''
  - add_host:
      name: "{{ item }}"
      groups: oo_list_hosts
      ansible_ssh_user: "{{ deployment_vars[deployment_type].ssh_user }}"
      ansible_ssh_host: "{{ hostvars[item].ansible_ssh_host | default(item) }}"
      ansible_become: "{{ deployment_vars[deployment_type].become }}"
      oo_public_ipv4: "{{ hostvars[item].openstack.public_v4 }}"
      oo_private_ipv4: "{{ hostvars[item].openstack.private_v4 }}"
    with_items: "{{ groups[scratch_group] | default([]) | difference(['localhost']) }}"
  - debug:
      msg: "{{ hostvars | oo_select_keys(groups[scratch_group] | default([])) | oo_pretty_print_cluster('meta-') }}"
```

##### Expected Result

Playbook completes without error.

##### Additional Information
See command in steps to reproduce.
CentOS Linux release 7.2.1511 (Core)
(inventory created by openstack cluster creator)

**Comment by dougbtv** [comment]
Secondarily, I checked out the previous commit, 769274f376ed189d74e9684e126c17f6ddd3d4ff

And I was able to have a successful run.

I'll later investigate what is missing for host variables that isn't provided.

**Comment by lhuard1A** [comment]
@dougbtv : Sorry for this regression. Should be fixed with #2858.
